modules/deploy/pages/deployment-option/cloud/aws-privatelink.adoc

```diff
@@ -0,0 +1,393 @@
++
+--
+- `name`
+- `cidr_block`
+- `aws_region`
+--
++
+[,bash]
+----
+REGION=<aws_region>
+
+NETWORK_POST_BODY=`cat << EOF
+{
+    "cloud_provider": "CLOUD_PROVIDER_AWS",
+    "cluster_type": "TYPE_BYOC",
+    "name": "<my-private-link-network>",
+    "cidr_block": "<10.0.0.0/20>",
+    "namespace_id": "$NAMESPACE_ID",
+    "region": "$REGION"
+}
+EOF`
+
+NETWORK_ID=`curl -vv -X POST \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $AUTH_TOKEN" \
+    -d "$NETWORK_POST_BODY" $PUBLIC_API_ENDPOINT/v1beta1/networks | jq .metadata.network_id`
+
+echo $NETWORK_ID
+----
++
+Wait for the network to be ready before creating the cluster in the next step. You can check the state of the network by calling https://docs.api.redpanda.com/#get-/v1beta1/networks/-id-[`GET /v1beta1/networks/{id}`]. You can create the cluster when the state is `STATE_READY`.
+
+. Create a new cluster with the endpoint service enabled by calling https://docs.api.redpanda.com/#post-/v1beta1/clusters[`POST /v1beta1/clusters`].
++
+In the example below, make sure to set your own values for the following fields:
++
+--
+- `zones`: for example, `"usw2-az1","usw2-az2","usw2-az3"`
+- `type`: `TYPE_DEDICATED` for Dedicated Cloud clusters, or `TYPE_BYOC` for BYOC clusters
+- `tier`: for example, `tier-1-aws-v2-arm`
+- `name`
+- `allowed_principals`: Amazon Resource Names (ARNs) for the AWS principals allowed to access the endpoint service. For example, for all principals in an account, use `arn:aws:iam::account_id:root`. See https://docs.aws.amazon.com/vpc/latest/privatelink/configure-endpoint-service.html#add-remove-permission[Configure an endpoint service] for details.
+--
++
+[,bash]
+----
+CLUSTER_POST_BODY=`cat << EOF
+{
+    "cloud_provider": "CLOUD_PROVIDER_AWS",
+    "connection_type": "CONNECTION_TYPE_PRIVATE",
+    "name": "<my-private-link-cluster>",
+    "namespace_id": "$NAMESPACE_ID",
+    "network_id": $NETWORK_ID,
+    "region": "$REGION",
+    "zones": [ <zones> ],
+    "throughput_tier": "<tier>",
+    "type": "<type>",
+    "private_link": {
+        "enabled": true,
+        "aws": {
+        "allowed_principals": ["<principal_1>","<principal_2>", ...]
+        }
+    }
+}
+EOF`
+
+CLUSTER_ID=`curl -vv -X POST \
+    -H "Content-Type: application/json" \
+    -H "Authorization: Bearer $AUTH_TOKEN" \
+    -d "$CLUSTER_POST_BODY" $PUBLIC_API_ENDPOINT/v1beta1/clusters | jq .metadata.cluster_id | sed 's/"//g'`
+
+echo $CLUSTER_ID
+----
++
+**BYOC clusters:** Check that the cluster operation is completed by calling https://docs.api.redpanda.com/#get-/v1beta1/operations/-id-[`GET /v1beta1/operations/{id}`], and passing the operation ID returned from the Create Cluster call.
++
+When the Create Cluster operation is completed (`STATE_COMPLETED`), execute the following `rpk cloud` command to finish creating your BYOC cluster:
++
+[,bash]
+----
+rpk cloud byoc aws apply --redpanda-id=$CLUSTER_ID
```

Comment:
@kbatuigas There was a Slack converstation re the `rpk cloud byoc aws apply` command in #help-cloud today: https://redpandadata.slack.com/archives/C0191NM4WU9/p1708719388376429
The gist of it is that this command does not appear in the rpk commands in the docs. The SEs shared that they did not know what was necessary for this step to work...